<launch>
    <arg name="cam_num_left" default="1" />
    <arg name="cam_num_right" default="2" />
    <arg name="cam_name_left" default="left" />
    <arg name="cam_name_right" default="right" />
    <arg name="width" default="1640" />
    <arg name="height" default="1232" />
    <arg name="enable_compression" default="0" />
    <arg name="enable_display" default="0" />

    <group ns="stereo">
        <param name="width" value="$(arg width)" />
        <param name="height" value="$(arg height)" />

	    <group ns="left">
            <node pkg="pi_cam_driver" type="cam6.py" name="$(arg cam_name_left)" >
                <param name="cam_num" value="$(arg cam_num_left)" />
                <param name="camera_info_url" value="package://settings/cam_left.yaml" />
                <param name="cam_name" value="$(arg cam_name_left)" />
                <param name="frame_id" value="stereo/left" />
            </node>

            <node pkg="image_proc" type="image_proc" name="$(arg cam_name_left)_rect" />
        </group>

        <group ns="right">
            <node pkg="pi_cam_driver" type="cam6.py" name="$(arg cam_name_right)" >
                <param name="cam_num" value="$(arg cam_num_right)" />
                <param name="camera_info_url" value="package://settings/cam_right.yaml" />
                <param name="cam_name" value="$(arg cam_name_right)" />
                <param name="frame_id" value="stereo/right" />
            </node>

            <node pkg="image_proc" type="image_proc" name="$(arg cam_name_left)_rect" />
        </group>

        <node pkg="image_transport" type="republish" name="republish"
           args="compressed" if="$(arg enable_compression)">
            <remap from="in" to="image_raw" />
            <remap from="out" to="image_raw_out" />
        </node>

        <node name="cam_view" pkg="image_view" type="image_view" respawn="false" output="screen" if="$(arg enable_display)" >
            <remap from="image" to="/camera/image_raw"/>
        </node>
    </group>

</launch>
